AutoRest/Generators/Ruby/Azure.Ruby/AzureRubyCodeGenerator.cs

```diff
@@ -58,6 +66,78 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
             AzureExtensions.NormalizeAzureClientModel(serviceClient, Settings, CodeNamer);
             CorrectFilterParameters(serviceClient);
             base.NormalizeClientModel(serviceClient);
+            AddRubyPageableMethod(serviceClient);
+            ApplyPagination(serviceClient);
+
+        }
+
+        /// <summary>
+        /// Adds method to use for autopagination
+        /// </summary>
+        /// <param name="serviceClient">The service client.</param>
+        private void AddRubyPageableMethod(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException("serviceClient");
+            }
+
+            for (int i = 0; i < serviceClient.Methods.Count; i++)
+            {
+                var method = serviceClient.Methods[i];
+                if (method.Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                {
+                    var pageableExtension = JsonConvert.DeserializeObject<Model.PageableExtension>(method.Extensions[AzureExtensions.PageableExtension].ToString());
+                    if (!method.Extensions.ContainsKey("nextLinkMethod") && !string.IsNullOrWhiteSpace(pageableExtension.NextLinkName)) 
+                    {
+                        serviceClient.Methods.Insert(i, (Method)method.Clone());
+                        method.Name = CodeNamer.GetMethodName(method.Name);
+                        i++;
```

Comment:
instead of using i as pointer to manipulate methods at specific location in list, why don't we just create new list of methods object with putting new methods and replacing them at the end? Or i might be mistaken on how 'i' is being used